### PR TITLE
add metamask authentification function

### PR DESCRIPTION
**Pull request by sashameniakin**
added login/logout with Metamask, user session data, Metamask address in the header

### CODE REVIEW

**Review by stefankorder — CHANGES_REQUESTED** [review]
Good so far! Please take a look at my comments. You should take also a look about how you call your components. Components with no content between opening and closing tag could be also called on a self-closing way.

File: components/Layout.js [hunk 1]
```diff
@@ -2,14 +2,14 @@ import Navbar from "./Navbar";
...
 
 export default function Layout({children}) {
-  const {path} = useRouter();
+  /* const {path} = useRouter(); */
 
   return (
     <>
-      {path === "/home" ? <Header /> : <></>}
+      <Header />
       <StyledMain>{children}</StyledMain>
```
[the remark this answers is not in the document]
Comment (sashameniakin) [reply components/Layout.js:14]: done!

File: components/Layout.js [hunk 2]
```diff
@@ -2,14 +2,14 @@ import Navbar from "./Navbar";
 import Header from "./Header";
 import styled from "styled-components";
 
-import {useRouter} from "next/router";
+/* import {useRouter} from "next/router"; */
 
 export default function Layout({children}) {
-  const {path} = useRouter();
+  /* const {path} = useRouter(); */
```
[the remark this answers is not in the document]
Comment (sashameniakin) [reply components/Layout.js:8]: done!

File: components/Layout.js [hunk 3]
```diff
@@ -2,14 +2,14 @@ import Navbar from "./Navbar";
 import Header from "./Header";
 import styled from "styled-components";
 
-import {useRouter} from "next/router";
+/* import {useRouter} from "next/router"; */
```
[the remark this answers is not in the document]
Comment (sashameniakin) [reply components/Layout.js:5]: done!

File: components/Header.js [hunk 4]
```diff
@@ -24,3 +57,16 @@ const StyledHeader = styled.header`
 const StyledSection = styled.section`
   margin-top: 10%;
 `;
+const StyledButton = styled.button`
+  top: 0px;
```
[the remark this answers is not in the document]
Comment (sashameniakin) [reply components/Header.js:61]: done!

File: components/Header.js [hunk 5]
```diff
@@ -1,14 +1,47 @@
...
+        destination: "/login",
+        permanent: false,
+      },
+    };
+  }
+}
+
 const StyledHeader = styled.header`
   position: fixed;
   width: 100%;
```
[the remark this answers is not in the document]
Comment (sashameniakin) [reply components/Header.js:47]: when i delete it, the header is 1/3 window's width

File: components/Header.js [hunk 6]
```diff
@@ -1,14 +1,47 @@
...
     <>
-      <StyledHeader></StyledHeader>
+      <StyledHeader>
+        <div></div>
+        <StyledAddress>{metamaskAddress}</StyledAddress>
+        <StyledButton onClick={() => signOut({redirect: "/login"})}>
+          <StyledImage alt="signout button" src={SignOut}></StyledImage>
+        </StyledButton>
+        <StyledButton onClick={() => openPopup()}>
+          <StyledImage alt="bookmark" src={Bookmark}></StyledImage>
```
[the remark this answers is not in the document]
Comment (sashameniakin) [reply components/Header.js:24]: done!

File: components/Header.js [hunk 7]
```diff
@@ -1,14 +1,47 @@
...
+
+  function openPopup() {
+    setGlobalState("openPopup", true);
+  }
+
   return (
     <>
-      <StyledHeader></StyledHeader>
+      <StyledHeader>
+        <div></div>
```
[the remark this answers is not in the document]
Comment (sashameniakin) [reply components/Header.js:18]: forgot to delete

File: components/Header.js [hunk 8]
```diff
@@ -1,14 +1,47 @@
...
+        <StyledAddress>{metamaskAddress}</StyledAddress>
+        <StyledButton onClick={() => signOut({redirect: "/login"})}>
+          <StyledImage alt="signout button" src={SignOut}></StyledImage>
+        </StyledButton>
+        <StyledButton onClick={() => openPopup()}>
+          <StyledImage alt="bookmark" src={Bookmark}></StyledImage>
+        </StyledButton>
+      </StyledHeader>
       <StyledSection></StyledSection>
```
[the remark this answers is not in the document]
Comment (sashameniakin) [reply components/Header.js:27]: deleted!

**Review by stefankorder — APPROVED** [review]
Well done! About the header element take a look at my comment.

File: components/Header.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (sashameniakin) [reply components/Header.js:45]: i will try ti find it out

**Pull request merged**